xarray/core/parallel.py

```diff
@@ -293,7 +291,7 @@ def _wrapper(
             )
 
         # check that index lengths and values are as expected
-        for name, index in result.xindexes.items():
+        for name, index in result.indexes.items():
```

Comment:
I'm scratching my head at trying to find how/where to fix the failing tests when using `xarray.PandasIndex` instead of `pd.Index`.

Error messages suggest that indexes are treated as arrays [here](https://github.com/pydata/xarray/blob/297756319738938e7431b829ca7390e7f2a79fd2/xarray/core/parallel.py#L298), [here](https://github.com/pydata/xarray/blob/297756319738938e7431b829ca7390e7f2a79fd2/xarray/core/parallel.py#L572) and maybe at other places too.

I have reverted `xindexes` to `indexes` everywhere in `map_blocks` so tests should pass now, but this is a temporary fix. Any help on this would be greatly appreciated! `map_blocks` is a huge function that I'm not very familiar with.

Reply:
#5670 fixes the two places you identified.

Reply:
Thanks for #5670 @dcherian! I merged it too quickly so I added a release note item in this PR.

I also had to temporarily fix the (xarray) indexes passed as coordinates to the resulting `Dataset`. All `map_blocks` tests should be passing using `xindexes` now.